File: src/main/python/bruteforce/숫자야구2503.py
N = int(input())
hint_list = [list(map(int,(input().split()))) for _ in range(N)]
# 123 1 1
# 356 1 0
# 327 2 0
# 489 0 1
answer = 0

for a in range(1,10):
  for b in range(1,10):
    for c in range(1,10):
      if a != b and b != c and c != a:
        count = 0
        for hint in hint_list:
          strike_count = 0
          ball_count = 0
          hint_input = hint[0] #123
          hint_100 = hint_input // 100  # 100의 자리수
          hint_10 = (hint_input % 100) // 10  # 10의 자리수
          hint_1 = (hint_input % 10) // 1 # 1의 자리수
          if (a == hint_100):
            strike_count+=1
          if (b == hint_10):
            strike_count+=1
          if (c== hint_1):
            strike_count+=1
          if (a == hint_10 or a== hint_1):
            ball_count+=1
          if b == hint_100 or b == hint_1:
            ball_count+=1
          if c== hint_100 or c == hint_10:
            ball_count+=1
          if strike_count == hint[1] and ball_count == hint[2]:
            count+=1
        if count == N:
          answer +=1
          count = 0

print(answer)

File: src/main/java/org/snust/bruteforce/숫자야구2503.java
package org.snust.bruteforce;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;
import java.util.StringTokenizer;

public class 숫자야구2503 {
  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st;
    int N = Integer.parseInt(br.readLine());
    int answer = 0;
    List<List<Integer>> hintList = new ArrayList<>(); // {}
    for (int i = 0; i < N; i++) {
      st = new StringTokenizer(br.readLine(), " ");
      List<Integer> list = new ArrayList<>();
      for (int j = 0; j < 3; j++) {
        list.add(Integer.parseInt(st.nextToken()));
      }
      hintList.add(list);
    }

    for (int a = 1; a < 10; a++) {
      for (int b = 1; b < 10; b++) {
        for (int c = 1; c < 10; c++) {
          int count = 0;
          if (a == b | b == c | c == a) continue;
          for (List<Integer> hint : hintList) {
            int number = hint.get(0); // 123
            int number100 = number / 100;
            int number10 = (number % 100) / 10;
            int number1 = number % 10;
            int strike = hint.get(1);
            int ball = hint.get(2);
            int strikeCount = 0;
            int ballCount = 0;
            if (a == number100) strikeCount++;
            if (b == number10) strikeCount++;
            if (c == number1) strikeCount++;

            if (a == number10 | a == number1) ballCount++;
            if (b == number100 | b == number1) ballCount++;
            if (c == number100 | c == number10) ballCount++;

            if (strikeCount == strike & ballCount == ball) count++;

            if (count == N) {
              answer++;
              count = 0;
            }
          }
        }
      }
    }
    System.out.println(answer);
  }
}
